cmd/vspd/main.go

```diff
@@ -28,122 +30,143 @@ const maxVoteChangeRecords = 10
 const consistencyInterval = 30 * time.Minute
 
 func main() {
+	// Load config file and parse CLI args.
+	cfg, err := loadConfig()
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "loadConfig error: %v\n", err)
+		os.Exit(1)
+	}
+
+	vspd, err := newVspd(cfg)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "newVspd error: %v\n", err)
+		os.Exit(1)
+	}
+
 	// Run until an exit code is returned.
-	os.Exit(run())
+	os.Exit(vspd.run())
 }
 
-// run is the main startup and teardown logic performed by the main package. It
-// is responsible for parsing the config, creating dcrd and dcrwallet RPC clients,
-// opening the database, starting the webserver, and stopping all started
-// services when a shutdown is requested.
-func run() int {
+type vspd struct {
+	cfg         *config
+	log         slog.Logger
+	shutdownCtx context.Context
+	db          *database.VspDatabase
+	dcrd        rpc.DcrdConnect
+	wallets     rpc.WalletConnect
+}
 
-	// Load config file and parse CLI args.
-	cfg, err := loadConfig()
+// newVspd creates the essential resources required by vspd - a database, logger
+// and RPC clients - then returns an instance of vspd which is ready to be run.
+func newVspd(cfg *config) (*vspd, error) {
+
```

Comment:
Extra blank line. I'm kind of surprised the linter isn't complaining.  I saw several other cases of this, but only commenting once.

Reply:
Fixed